test/hotspot/jtreg/compiler/igvn/CmpDisjointButNonOrderedRangesLong.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8360561
+ * @summary Ranges can be proven to be disjoint but not orderable (thanks to unsigned range)
+ *          Comparing such values in such range with != should always be true.
+ * @library /test/lib /
+ * @run main compiler.igvn.CmpDisjointButNonOrderedRangesLong
+ */
+package compiler.igvn;
+
+import compiler.lib.ir_framework.*;
+
+public class CmpDisjointButNonOrderedRangesLong {
+    static boolean bFld;
+    static double dFld1;
+    static double dFld2;
+
+    public static void main(String[] strArr) {
+        TestFramework.runWithFlags("-Xcomp", "-XX:CompileCommand=compileonly,compiler.igvn.CmpDisjointButNonOrderedRangesLong::test");
```

Comment:
Supplying `-Xcomp` will skip IR-verification:

https://github.com/openjdk/jdk/blob/fc4755535d61c2fd4d9a2c9a673da148f742f035/test/hotspot/jtreg/compiler/lib/ir_framework/README.md?plain=1#L136-L141

You can emulate `-Xcomp` behavior with `@Warmup(0)`.

Reply:
Good to know. Thank you for clearing that up for me.

Reply:
Indeed. I use it here to prevent profiling from removing an actually impossible path with a trap, because bad things happen in a dead path. It's not the first time I use `Xcomp` for that, and there are other ways (like setting a maximum on the number of traps per method, or disabling the warmup (and so profiling) in IR framework execution). That was discussed in some other PR without strong opinions or consensus on what would be the preferred way.